airflow/providers/microsoft/azure/hooks/wasb.py

```diff
@@ -173,8 +173,8 @@ def get_conn(self) -> BlobServiceClient:
                 # provided the DNS name without the https:// prefix.
                 # Azure storage account name can only be 3 to 24 characters in length
                 # https://learn.microsoft.com/en-us/azure/storage/common/storage-account-overview#storage-account-name
-                acc_name = account_url.split(".")[0][:24]
-                account_url = f"https://{acc_name}." + ".".join(account_url.split(".")[1:])
+                acc_name, acc_url = account_url.split(".", 1)
+                account_url = f"https://{acc_name[:24]}.{acc_url}"
```

Comment:
This logic would produce a URL with a trailing dot if the original URL does not contain any dots. But the original implementation already has this problem so I’m not sure if we want to fix it here.

Reply:
Since you mentioned it, I've suggested a fix.